src/google/Timeline.js

```diff
@@ -0,0 +1,109 @@
+    
+    Timeline.prototype.publish("backgroundColor", null, "html-color", "The background color for the main area of the chart. Can be either a simple HTML color string, for example: 'red' or '#00cc00'.",null,{tags:['Basic']});
+ 
+    /**
+     * Publish Params Unique To This Widget
+     */   
+    Timeline.prototype.publish("avoidOverlappingGridLines", true, "boolean", "Whether display elements (e.g., the bars in a timeline) should obscure grid lines. If false, grid lines may be covered completely by display elements. If true, display elements may be altered to keep grid lines visible.",null,{tags:['Basic']});
+
+    Timeline.prototype.publish("timelineColorByRowLabel", false, "boolean", "If set to true, colors every bar on the row the same. The default is to use one color per bar label.",null,{tags:['Basic']});
+    
+    Timeline.prototype.publish("timelineGroupByRowLabel", true, "boolean", "If set to false, creates one row for every dataTable entry. The default is to collect bars with the same row label into one row.",null,{tags:['Basic']});
+ 
+    Timeline.prototype.publish("timelineShowBarLabels", true, "boolean", "If set to false, omits bar labels. The default is to show them.", null,{tags:['Basic']});
+    
+    Timeline.prototype.publish("timelineShowRowLabels", true, "boolean", "If set to false, omits row labels. The default is to show them.",null,{tags:['Basic']});
+    
+    Timeline.prototype.publish("timelineSingleColor", null, "string", "Colors all bars the same. Specified as a hex value (e.g., '#8d8').",null,{tags:['Basic']});
+    
+    Timeline.prototype.getChartOptions = function () {
+        var retVal = [];
+        
+        retVal.avoidOverlappingGridLines = this.avoidOverlappingGridLines();
+        retVal.backgroundColor = this.backgroundColor();
+        retVal.timelineColorByRowLabel = this.timelineColorByRowLabel();
+        retVal.timelineGroupByRowLabel = this.timelineGroupByRowLabel();
+        retVal.timelineShowBarLabels = this.timelineShowBarLabels();
+        retVal.timelineShowRowLabels = this.timelineShowRowLabels();
+        retVal.timelineSingleColor = this.timelineSingleColor();
+        retVal.tooltipIsHtml = this.tooltipIsHtml();
+        retVal.tooltipTrigger = this.tooltipTrigger();
+
+        return retVal;
+    };
+
+    Timeline.prototype.enter = function (domNode, element) {
+        element.style("overflow", "hidden");
+        this._chart = new google.visualization[this._chartType](domNode);
+    };
+
+    Timeline.prototype.update = function (domNode, element) {
+        HTMLWidget.prototype.update.apply(this, arguments);
+       
+        this._chart.draw(this._data_google, this.getChartOptions());
+    };
+    Timeline.prototype.data = function (_) {
+        this._data = _;
```

Comment:
You need to test if arguments.length === 0 and return the data.  See: https://github.com/hpcc-systems/Visualization/blob/master/src/map/ChoroplethCountries.js#L44-L64 for an example of what I mean